File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int pL = 0;
        int pR = numbers.length-1;

        while (numbers[pL] + numbers[pR] != target) {
            if (numbers[pL] + numbers[pR] > target) pR--;
            if (numbers[pL] + numbers[pR] < target) pL++;
        }

        return new int[]{pL+1,pR+1};
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        lp, rp = 1, max(piles)

        while lp <= rp:
            mid = (lp + rp) // 2
            time = 0
            for pile in piles:
                time += math.ceil(pile / mid)

            # if eating time < target (eating too fast), slow down
            if time <= h:
                rp = mid - 1
            # if eating time > target (eating too slow), speed up
            elif time > h:
                lp = mid + 1

        return lp